packages/angular/cli/models/parser.ts

```diff
@@ -270,30 +284,73 @@ export function parseArguments(args: string[], options: Option[] | null): Argume
   const ignored: string[] = [];
   const errors: string[] = [];
 
-  for (let arg = args.shift(); arg !== undefined; arg = args.shift()) {
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+
     if (arg == '--') {
       // If we find a --, we're done.
-      leftovers.push(...args);
+      leftovers.push(...args.slice(i + 1));
       break;
     }
 
     if (arg.startsWith('--')) {
-      _assignOption(arg, args, options, parsedOptions, positionals, leftovers, ignored, errors);
+      const consumedNextArg = _assignOption({
+        arg,
+        nextArg: args[i + 1],
+        options,
+        parsedOptions,
+        leftovers,
+        ignored,
+        errors,
+      });
+
+      if (consumedNextArg) {
+        i++;
+      }
+
     } else if (arg.startsWith('-')) {
       // Argument is of form -abcdef.  Starts at 1 because we skip the `-`.
-      for (let i = 1; i < arg.length; i++) {
-        const flag = arg[i];
+      for (let j = 1; j < arg.length; j++) {
+        const flag = arg[j];
         // If the next character is an '=', treat it as a long flag.
-        if (arg[i + 1] == '=') {
-          const f = '--' + flag + arg.slice(i + 1);
-          _assignOption(f, args, options, parsedOptions, positionals, leftovers, ignored, errors);
+        if (arg[j + 1] == '=') {
+          const f = '--' + flag + arg.slice(j + 1);
+
+          const consumedNextArg = _assignOption({
+            arg: f,
```

Comment:
For this usage of _assignOption we pass in a computed value rather than a value from the args array